python/ray/tune/experiment/trial.py

```diff
@@ -536,7 +536,7 @@ def remote_experiment_path(self) -> str:
 
     @property
     def local_experiment_path(self) -> str:
-        return self.storage.experiment_local_path
+        return self.storage.experiment_local_staging_path
```

Comment:
Should we delete all of these `Trial` properties and just always access paths through the `storage`? Neither are really stable APIs.

Reply:
Same thing for the `Experiment` class

Reply:
Yeah, the naming of these properties are kind of confusing. We can deprecate these properties and take `StorageContext` as ground truth.

Reply:
Ok, I'll do it in a followup PR (doesn't have to make branch cut).